### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by pmjohann**
I found the ability to run cryptomator cli containerized very useful, so I thought I'd share the Dockerfile I wrote as a contribution.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.12.0
+
+ARG CRYPTOMATOR_CLI_VERSION=0.4.0
+
+RUN adduser -D cryptomator && \
+    apk add --no-cache curl openjdk11-jre-headless && \
+    curl -sSL https://github.com/cryptomator/cli/releases/download/$CRYPTOMATOR_CLI_VERSION/cryptomator-cli-$CRYPTOMATOR_CLI_VERSION.jar > /usr/bin/cryptomator.jar && \
```
Comment (SuperSandro2000) [inline Dockerfile:7]: Why are you not using wget which is included by default?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,36 @@ java -jar cryptomator-cli-x.y.z.jar \
 # you can now mount http://localhost:8080/demoVault/
 ```
 
+## Using as a docker image
+
+*Bridge networking with port forward:*
```
Comment (SuperSandro2000) [inline README.md:30]: Without disabling icc anyone can access that port.

File: Dockerfile (hunk 1 above)
Comment (pmjohann) [reply Dockerfile]: Makes perfect sense, I am just more familiar with curl in-depth. [Already replaced curl with wget in my fork](https://github.com/pmjohann/cli/commit/14d10739f6a0cb04e457dd67558d83d982729f48), I'll start a new pull request with the changes as soon as I updated the readme as well related to your other comment.

Comment (SuperSandro2000) [reply Dockerfile]: This PR gets automatically updates if you change it in the fork.

File: README.md (hunk 2 above)
Comment (pmjohann) [reply README.md:30]: Yes, but this scenario is just for testing and understanding how the container works.
I use this specifically in a private network where another container running rclone pulls the vault from the cloud, cryptomator decrypts the vault and serves it to another container which grabs the content via WebDAV. Here one can define the --bind address to the container IP itself which the other container would access it on in a private network environment.
I will also update the README.md to indicate this.

**Comment by pmjohann** [comment]
I addressed the two issues which came up in the conversation above, feel free to make any other suggestion / requirement.

File: README.md [hunk 3]
```diff
@@ -25,6 +25,42 @@ java -jar cryptomator-cli-x.y.z.jar \
...
+
+### !!! WARNING: This approach should only be used to test the containerized approach, not in production.
```
Comment (overheadhunter) [inline README.md:32]: Just a little markdown semantics improvement. Also, use emoji to highlight the warning

```suggestion
### Bridge networking with port forward:

:warning: **WARNING: This approach should only be used to test the containerized approach, not in production.** :warning:
```

File: README.md [hunk 4]
```diff
@@ -25,6 +25,42 @@ java -jar cryptomator-cli-x.y.z.jar \
...
+    -v /path/to/differentVault:/vaults/differentVault \
+    -v /path/to/fileWithPassword:/passwordFile \
+    --bind 0.0.0.0 --port 8080 \
+    cryptomator/cli \
+    --vault demoVault=/vaults/vault --password demoVault=topSecret \
+    --vault otherVault=/vaults/differentVault --passwordfile otherVault=/passwordFile
+# you can now mount http://localhost:8080/demoVault/
+```
+
+*Host networking:*
```
Comment (overheadhunter) [inline README.md:50]:
```suggestion
### Host networking:
```

**Comment by overheadhunter** [comment]
Also, the part about mounting might should get its own `##` heading, as it doesn't strictly belong just to the docker section.

**Comment by pmjohann** [comment]
Hi @overheadhunter , I implemented your suggestions in my branch, however GItHub still tells me Changes requested, although I marked them as resolved. Maybe I overlooked something?

Let me know if you need me to make other amendments, and keep up the good work! ;)

**Comment by overheadhunter** [comment]
> however GItHub still tells me Changes requested

Seems like the state only changes after explicitly approving the changes. Which I did now 😉

**Comment by overheadhunter** [comment]
Thanks for your contribution!

**Pull request merged**